pkg/security/certManager/general/certManager.go

```diff
@@ -68,27 +74,27 @@ func New(config Config, fileWatcher *fsnotify.Watcher, logger log.Logger) (*Cert
 		verifyClientCertificate: verifyClientCertificate,
 		logger:                  logger,
 	}
-	err := c.loadCAs()
+	_, err := c.loadCAs()
 	if err != nil {
 		return nil, err
 	}
-	err = c.loadCerts()
+	_, err = c.loadCerts()
 	if err != nil {
 		return nil, err
 	}
 	for _, ca := range config.CAPool {
-		if err := c.fileWatcher.Add(filepath.Dir(ca)); err != nil {
-			return nil, fmt.Errorf("cannot watch CAPool directory(%v): %w", filepath.Dir(ca), err)
+		if err := c.fileWatcher.Add(ca); err != nil {
+			return nil, fmt.Errorf("cannot watch CAPool(%v): %w", ca, err)
 		}
 	}
 	if config.CertFile != "" {
-		if err := c.fileWatcher.Add(filepath.Dir(config.CertFile)); err != nil {
-			return nil, fmt.Errorf("cannot watch CertFile directory(%v): %w", filepath.Dir(config.CertFile), err)
+		if err := c.fileWatcher.Add(config.CertFile); err != nil {
+			return nil, fmt.Errorf("cannot watch CertFile(%v): %w", config.CertFile, err)
```

Comment:
Shouldn't we remove the previously registered observed paths before returning with error?